test/ffi_native_test/build_test_dylib.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
```

Comment:
If we want to run the native code, can we extend the `native_test` instead of copy pasting code? Adding a second `config.yaml` there.

Also, running the code will not work on all OSes. Only Windows and MacOS will do `DynamicLibrary.open` with global symbol visibility, the test will fail on Linux. https://github.com/dart-lang/sdk/issues/50105#issuecomment-1441764283 `@Native`s are supposed to be used with the native assets feature https://github.com/dart-lang/sdk/issues/50565 which provides the mapping from `@Native`s to the dylib path. Maybe land the PR with only expect code gen tests instead of running the test. We already have a codegen test in test/example_tests/ffinative_example_test.dart.

Reply:
I wasn't exactly sure if you only wanted to update `test/example_tests/ffinative_example_test.dart` or also move the codegen test into `test/native_test/native_test.dart` so I did both. If that's unnecessary duplication I can just delete the latter.

Reply:
The latter is unnecessary, it's just testing the same thing right?

Reply:
Correct, I'll remove.